examples/cifar10_cnn_capsule.py

```diff
@@ -0,0 +1,222 @@
+"""Train a simple CNN-Capsule Network on the CIFAR10 small images dataset.
+
+Without Data Augmentation:
+It gets to 75% validation accuracy in 10 epochs,
+and 79% after 15 epochs, and overfitting after 20 epochs
+
+With Data Augmentation:
+It gets to 75% validation accuracy in 10 epochs,
+and 79% after 15 epochs, and 83% after 30 epcohs.
+In my test, highest validation accuracy is 83.79% after 50 epcohs.
+
+This is a fast Implement, just 20s/epcoh with a gtx 1070 gpu.
+"""
+
+from __future__ import print_function
+from keras import backend as K
+from keras.engine.topology import Layer
+from keras import activations
+from keras import utils
+from keras.datasets import cifar10
+from keras.models import Model
+from keras.layers import *
+from keras.preprocessing.image import ImageDataGenerator
+
+
+# the squashing function.
+# we use 0.5 in stead of 1 in hinton's paper.
+# if 1, the norm of vector will be zoomed out.
+# if 0.5, the norm will be zoomed in while original norm is less than 0.5
+# and be zoomed out while original norm is greater than 0.5.
+def squash(x, axis=-1):
+    s_squared_norm = K.sum(K.square(x), axis, keepdims=True) + K.epsilon()
+    scale = K.sqrt(s_squared_norm) / (0.5 + s_squared_norm)
+    return scale * x
+
+
+# define our own softmax function instead of K.softmax
+# because K.softmax can not specify axis.
```

Comment:
That's something we should fix in the Keras backend.

Reply:
Wasn't it implemented in https://github.com/keras-team/keras/pull/8841?